app/src/main/java/org/mozilla/webmaker/web/javascript/WebAppInterface.java

```diff
@@ -229,7 +228,7 @@ public void setView(final String url, final String routeData) {
         if (activity == null) return;
 
         if (routeData != null) {
-            MemStorage.sharedStorage().put(ROUTE_KEY, routeData);
+            api.queue(ROUTE_KEY, routeData);
```

Comment:
if we don't add a `setPayload` method, this will need to make sure we clear first, too. Otherwise, this should become `setPayload`.